### PR TITLE
3942 fix invalid credentials msg UI

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #3942 .

Any suggestion about fixing the logic is much appreciated.

Sorting is disabled but I will try to do that as well after we merge this PR.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [ ] Do all unit tests pass?
* [ ] Do all end-to-end tests pass?
* [x] Any other testing performed?
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [x] If applicable, add screenshots or log transcripts of the feature working

image

### CODE REVIEW

**Comment by VakarisZ** [comment]
I'm seeing things like: 
image

But anyway, I think these error messages could pretty much just be the contents without the index. The actual solution is dynamic validation. We can discuss what we want to do during the release planning, I've made issues https://github.com/guardicore/monkey/issues/3974 and https://github.com/guardicore/monkey/issues/3975

**Comment by ilija-lazoroski** [comment]
> I'm seeing things like: image
> 
> But anyway, I think these error messages could pretty much just be the contents without the index. The actual solution is dynamic validation. We can discuss what we want to do during the release planning, I've made issues #3974 and #3975

For the above message, you need to sync your island dependencies.

File: monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js [hunk 1]
```diff
@@ -398,17 +399,43 @@ class ConfigurePageComponent extends AuthComponent {
   }
 
   sendCredentials() {
+    let islandCredentials = formatCredentialsForIsland(this.state.credentials.credentialsData);
+    let formattedCredentials = JSON.stringify(islandCredentials);
+    let formCredentialsIsland = {};
+    let credentialsDataId = {};
+    this.state.credentials.credentialsData.forEach((stateCredentials, stateCredentialsIndex) => {
+        formCredentialsIsland[stateCredentials.id] = formatCredentialsForIsland([stateCredentials]);
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js:407]: Didn't we agree on modifying and only calling this function once per row, and storing that data to reuse here?

File: monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js [hunk 2]
```diff
@@ -398,17 +399,43 @@ class ConfigurePageComponent extends AuthComponent {
   }
 
   sendCredentials() {
+    let islandCredentials = formatCredentialsForIsland(this.state.credentials.credentialsData);
+    let formattedCredentials = JSON.stringify(islandCredentials);
+    let formCredentialsIsland = {};
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js:404]: What does this object contain exactly? The name isn't very helpful.

File: monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js [hunk 3]
```diff
@@ -398,17 +399,43 @@ class ConfigurePageComponent extends AuthComponent {
...
         .then(res => {
           if (!res.ok) {
+            res.json().then((result) => {
+              let errorsId = {};
+              for(var credentialsId in formCredentialsIsland){
+                errorsId[credentialsId] = [];
+                formCredentialsIsland[credentialsId].forEach((formCredentials, _) => {
+                  let indexes = islandCredentials.map((islandCredentialsItem, islandCredentialsIndex) => JSON.stringify(formCredentials) === JSON.stringify(islandCredentialsItem) ? islandCredentialsIndex : -1)
+                  .filter(islandCredentialsIndex => islandCredentialsIndex !== -1);
+                  errorsId[credentialsId].push(indexes[0]);
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js:428]: Why are you only pushing `indexes[0]` to the array?

File: monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js [hunk 4]
```diff
@@ -398,17 +399,43 @@ class ConfigurePageComponent extends AuthComponent {
...
+                });
+              }
+              this.setState({
+                credentialsErrors: result.errors.map(error => {
+                  let credentialsErrorId = Object.entries(errorsId).find(([_, credentialsId]) => credentialsId.includes(error.index));
+                  let actualRowIndex = credentialsDataId[credentialsErrorId[0]] + 1;
+                  return `Credentials Row #${actualRowIndex}: ${error.message.split(', ').pop()}`;
```
Comment (shreyamalviya) [inline monkey/monkey_island/cc/ui/src/components/pages/ConfigurePage.js:435]:
```suggestion
                  return `Row #${actualRowIndex}: ${error.message.split(', ').pop()}`;
```

**Review by shreyamalviya — CHANGES_REQUESTED** [review]

**Pull request closed without merging**